Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -404,6 +404,39 @@ std::optional<SimpleRange> AccessibilityObject::misspellingRange(const SimpleRan
     return std::nullopt;
 }
 
+bool AccessibilityObject::isNodeForComposition(const Editor& editor) const
+{
+    auto* compositionNode = editor.compositionNode();
+    auto* cache = axObjectCache();
+    if (!compositionNode || !cache)
+        return false;
+
+    auto* compositionObject = cache->textCompositionObjectForNode(*compositionNode);
+    if (!compositionObject)
+        return false;
+
+    return compositionObject->objectID() == objectID();
```

Comment:
We could write this a little more simply like so:

```
if (auto* compositionObject = cache->textCompositionObjectForNode(*compositionNode))
    return compositionObject->objectID() == objectID();
return false;
```

Reply:
suggestion applied!